lib/salus/scanners/osv/go_osv.rb

```diff
@@ -0,0 +1,124 @@
+require 'salus/scanners/osv/base'
+
+module Salus::Scanners::OSV
+  class GoOSV < Base
+    class SemVersion < Gem::Version; end
+    class SemDependency < Gem::Dependency; end
+
+    EMPTY_STRING = "Not Found".freeze
+    DEFAULT_SOURCE = "https://osv.dev/list".freeze
+    DEFAULT_SEVERITY = "LOW".freeze
+
+    def should_run?
+      @repository.go_sum_present?
+    end
+
+    def find_dependencies
```

Comment:
Let's DRY up our code here.  We have existing language dependency reporters - each of which has logic for parsing the dependencies present.  I'd like to see us have those reporters and these OSV scanners use a common parsing method.

For example:
https://github.com/coinbase/salus/blob/master/lib/salus/scanners/report_go_dep.rb

Let's refactor as needed so we can have one common parser for each package manager.

Reply:
One of the options I am thinking is following the similar method to what we use, adding a parse_go_sum file bin/ folder.
This way both report_go_dep and go_osv can call that to get dependencies.